packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -4706,6 +4706,135 @@ class Wrap extends MultiChildRenderObjectWidget {
 ///  * [CustomMultiChildLayout], which uses a delegate to position multiple
 ///    children.
 ///  * The [catalog of layout widgets](https://flutter.dev/widgets/layout/).
+///
+/// {@tool snippet --template=freeform}
+///
+/// This example uses the [Flow] widget to create a menu that opens and closes
+/// as it is interacted with. The [FlowDelegate] uses the [AnimationController]
+/// to paint the children of the [Flow].
+///
+/// ![A menu created using the Flow widget opens and closes.](https://flutter.github.io/assets-for-api-docs/assets/widgets/flow_menu.mp4)
+///
+/// ```dart main
+/// import 'package:flutter/material.dart';
+///
+/// void main() => runApp(FlowApp());
+///
+/// class FlowApp extends StatelessWidget {
+///   @override
+///   Widget build(BuildContext context) {
+///     return MaterialApp(
+///       home: Scaffold(
+///         appBar: AppBar(
+///           title: const Text('Flow Example'),
+///         ),
+///         body: FlowMenu(),
+///       ),
+///     );
+///   }
+/// }
+///
+/// class FlowMenu extends StatefulWidget {
+///   @override
+///   _FlowMenuState createState() => _FlowMenuState();
+/// }
+///
+/// class _FlowMenuState extends State<FlowMenu>
+///   with SingleTickerProviderStateMixin {
+///   AnimationController menuAnimation;
+///   int active = 1;
+///   final Map<int, IconData> menuItems = <IconData>[
+///     Icons.home,
+///     Icons.new_releases,
+///     Icons.notifications,
+///     Icons.settings,
+///     Icons.menu,
+///   ].asMap();
+///
+///   @override
+///   void initState() {
+///     super.initState();
+///     menuAnimation = AnimationController(
+///       lowerBound: 1,
+///       upperBound: 1000,
+///       duration: const Duration(milliseconds: 250),
+///       vsync: this,
+///     );
+///   }
+///
+///   Widget buildItem(int k, IconData v) {
+///     return GestureDetector(
+///       onTap: () {
+///         if (k != 4) active = k;
```

Comment:
Why can 4 not be active?

Reply:
The item at the fourth index is the icon to open and close the menu. The active state is for the other button in the menu to indicate interaction. Will make this clearer. 😉